### PR TITLE
Upgrade Slurm from 20.11.5 to 20.11.7

### PR DESCRIPTION
**Pull request by enrico-usai**
+ Upgrade Slurm from 20.11.5 to 20.11.7
+ Use GitHub as repository to download source code (reverts 91cb7f7).

### CODE REVIEW

**Review by lukeseawalker — APPROVED** [review]

File: attributes/default.rb [hunk 1]
```diff
@@ -112,9 +112,9 @@
 default['cfncluster']['torque']['url'] = 'https://github.com/adaptivecomputing/torque/archive/6.1.2.tar.gz'
 # Slurm software
 default['cfncluster']['slurm_plugin_dir'] = '/etc/parallelcluster/slurm_plugin'
-default['cfncluster']['slurm']['version'] = '20.11.5'
-default['cfncluster']['slurm']['url'] = 'https://download.schedmd.com/slurm/slurm-20.11.5.tar.bz2'
-default['cfncluster']['slurm']['sha1'] = '201a28afe6f02a717fb348542878900cad4ccf13'
+default['cfncluster']['slurm']['version'] = '20.11.7'
```
Comment (rexcsn) [inline attributes/default.rb:115]: Please double check this. We use this version to interpolate the slurm tarball name and extracted name.
The downloaded tarball from github will have different name. And the extracted folder will be named slightly different as well.
I believe the version should be `20-11-7-1`, default downloaded tarball filename is `slurm-20-11-7-1.tar.gz`, and default extracted folder name is `slurm-slurm-20-11-7-1`.

Comment (enrico-usai) [reply attributes/default.rb:115]: Fixed thanks

Comment (enrico-usai) [reply attributes/default.rb]: Fixed thanks. Patch still under testing.

**Comment by demartinofra** [comment]
Verified hashes and diffs
```
❯ shasum --algorithm 1 ./slurm-slurm-20-11-7-1.tar.gz
ce927fbf2f7d5f908ed87c5d521abc696b9a2508  ./slurm-slurm-20-11-7-1.tar.gz
❯ shasum --algorithm 1 ./slurm-20.11.7.tar.bz2
a5212eed09c4cb164b5c9b6c76e13c7bbec798d1  ./slurm-20.11.7.tar.bz2
❯ diff <(find "./slurm-slurm-20-11-7-1/" -type f -exec md5sum {} + | sort -k 2 | sed 's/ .*\// /') <(find "./slurm-20.11.7/" -type f -exec md5sum {} + | sort -k 2 | sed 's/ .*\// /')
```

**Review by demartinofra — APPROVED** [review]

**Pull request merged**